### ISSUE TITLE
[DoctrineBridge] Middleware fails to handle binary values

### ISSUE THREAD

**Issue by geoffrey-brier** [symfony/symfony#46744]
### Symfony version(s) affected

\>= 5.4

### Description

Hello there,

I just upgraded my application today (I switched from doctrine bundle 2.6.3 to 2.7.0) which is in 6.1 and my tests detected something weird.

Here is a part of the stack trace :

```
      Behat\Testwork\Call\Exception\FatalThrowableError: Type error: Symfony\Bridge\Doctrine\Middleware\Debug\Query::setValue(): Argument #2 ($value) must be of type string|int|float|bool|null, resource given, called in vendor/symfony/doctrine-bridge/Middleware/Debug/Statement.php on line 48 in vendor/symfony/doctrine-bridge/Middleware/Debug/Query.php:55
      Stack trace:
      #0 vendor/symfony/doctrine-bridge/Middleware/Debug/Statement.php(48): Symfony\Bridge\Doctrine\Middleware\Debug\Query->setValue(5, Resource id #2538, 16)
      #1 vendor/doctrine/dbal/src/Statement.php(115): Symfony\Bridge\Doctrine\Middleware\Debug\Statement->bindValue(5, Resource id #2538, 16)
      #2 vendor/doctrine/orm/lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php(274): Doctrine\DBAL\Statement->bindValue(5, Resource id #2538, Object(Doctrine\DBAL\Types\BinaryType))
      #3 vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php(1129): Doctrine\ORM\Persisters\Entity\BasicEntityPersister->executeInserts()
      #4 vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php(426): Doctrine\ORM\UnitOfWork->executeInserts(Object(Doctrine\ORM\Mapping\ClassMetadata))
      #5 vendor/doctrine/orm/lib/Doctrine/ORM/EntityManager.php(398): Doctrine\ORM\UnitOfWork->commit(NULL)
      #6 var/cache/test/ContainerZdJq7y7/EntityManager_9a5be93.php(136): Doctrine\ORM\EntityManager->flush(NULL)
      #7 src/Services/NetworkIpv6Manager.php(163): ContainerZdJq7y7\EntityManager_9a5be93->flush()
      #8 src/Controller/Ipv6Controller.php(136): App\Services\NetworkIpv6Manager->splitIpv6Subnet(Object(App\Entity\ConfNetworkclass), '2001:bc8:3000:1...')

```

After digging a bit, I realized that [this commit](https://github.com/doctrine/DoctrineBundle/commit/9583584ca814f15c961ac6c25a042eb34bb8dbef) introduced a change in the query loggers (dbal logger -> middleware) which is the starting point of the issue. Indeed, it looks like the `Query` class does not handle binary values so that's why we end up with a fatal error.

### How to reproduce

Create an entity with a field of 'binary' type (I guess it would work for blob values as well) ... pushing further the thought I'm even wondering if the DateTime would be handled correctly but I'm far from being an expert on this one :thinking:

**Comment by marforon** [comment]
It should also accept arrays as an argument, for example array of ids.

**Comment by dmaicher** [comment]
@l-vo could this maybe be relaxed to `mixed`?

**Comment by l-vo** [comment]
@dmaicher, yes, using mixed should fix the problem. doctrine/dbal 4.x uses a mixed type for the value.  I'm going to suggest a fix.

**Comment by l-vo** [comment]
PR proposed. Actually, versions 5.4 and 6.0 are not affected.

**Comment by chetzel** [comment]
Hi, thank you for the fix for setValue(), but how about setParam() 11 lines above it (line 46 in current implementations of Query) - looks like the exact same issue. I know bindParam() is marked as deprecated, still current solutions rely on this behavior.

**Comment by l-vo** [comment]
@chetzel indeed, I don't know why I didn't fix this at the same time. Thank you for reporting. Fix proposed on #50268
